cuda_util/cuda_algorithm.cuh

```diff
@@ -0,0 +1,22 @@
+/**
+ * ENVISAT and ERS ASAR instrument focusser for QA4EO activity (c) by CGI Estonia AS
+ *
+ * ENVISAT and ERS ASAR instrument focusser for QA4EO activity is licensed under a
+ * Creative Commons Attribution-ShareAlike 4.0 International License.
+ *
+ * You should have received a copy of the license along with this
+ * work. If not, see http://creativecommons.org/licenses/by-sa/4.0/
+ */
+
+#pragma once
+
+#include <algorithm>
+
+namespace alus::cuda::algorithm {
+
+template <typename T>
+__device__ inline const T clamp(const T& v, const T& lo, const T& hi) {
+    return max(lo, min(hi, v));
```

Comment:
Not something I necessarily think is good or bad, just a fact of life with C++:

this implementation calls small trivial functions, min max - no issue with optimizations, but on debug builds it is slower than handwritten one:

https://godbolt.org/z/McTocrWez